src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -87,7 +121,65 @@ def ramp_fit(model, buffsize, save_opt, readnoise_2d, gain_2d,
         Object containing optional GLS-specific ramp fitting data for the
         exposure
     """
+    ramp_class = create_ramp_fit_class(model, dqflags)
+
+    return ramp_fit_internal(
+        ramp_class, buffsize, save_opt, readnoise_2d, gain_2d,
+        algorithm, weighting, max_cores, dqflags)
+
+
+def ramp_fit_internal(ramp_class, buffsize, save_opt, readnoise_2d, gain_2d,
+                      algorithm, weighting, max_cores, dqflags):
+    """
+    ramp_class: RampFitInternal
```

Comment:
A summary here of what this does would be helpful, with so many functions already similarly named.

Reply:
This function exists as a buffer to allow for passage of a data model to `ramp_fit`.  That function creates a class to containing only the needed information for ramp fitting then calls this function.  All code after this function call no longer depends on a data model.

Reply:
Ok.

Reply:
Yeah, this should have at least 1-line docstring before the summary of Parameters.

```
"""
Description of what this function does

Parameters
----------
ramp_class : RampFitInternal
...
```